### DISCUSSION TITLE
Is it possible to NOT do any query until someone types a search word?

### DISCUSSION THREAD

**Discussion by mikezimm** [microsoft-search/pnp-modern-search#1720] [Q&A]
One thing I found when using the pnp search webparts on our tenant is that when I first load the page, it takes a LONG time to load compared to our on premise search -it shows found 1.6 Million items.

I think it may be due to using filters which may be built up based on the returned results.

So what I'm wondering is if it's possible to use the google approach.

Just have the search box and verticals when the page loads.

Then only show results/refiners after someone does an actual search.

Or is it best practice to only show the first X results?  Just not sure how to best approach it because in a large enterprise, if you have refiners, you expect them to be complete.

Thanks!

**Comment by wobba** [comment]
Configure the search web part similar to this, and ensure you uncheck  "Use a default value". See https://microsoft-search.github.io/pnp-modern-search/usage/search-results/connections/

![image](https://user-images.githubusercontent.com/51104/149800597-5d9bc7cd-7f9b-44fb-b037-dac176683e30.png)

**Comment by mikezimm** [comment]
@wobba ,
It looks like I already had that unchecked..... This is what my settings show (first screenshot)
But the difference I have is I am also connecting to the Filters and Verticals webpart because I think that's what I need to make those things work right?

Screenshot #2 shows my Query Template (maybe that's the issue)

Screenshot #3 shows the page after it loads (5+ seconds and returns 1.58 Million results)

## My Settings:
![image](https://user-images.githubusercontent.com/49648086/149812952-5104af95-02b1-4b70-98bc-073d8db9f3a5.png)

## My Query Template:
![image](https://user-images.githubusercontent.com/49648086/149812994-9118b5d9-4acc-42f0-aceb-5e244a97349b.png)

## My loaded page (without typing any search words)
![image](https://user-images.githubusercontent.com/49648086/149813043-4591fe66-b07c-4ba4-90cf-4cf488b96812.png)

**Reply by mikezimm** [reply]
@wobba ,
NOTE:  I just did another test just seeing if it works.

I removed the * from the search box webpart and also the * from the query template.

Result:  Even worse, I actually got 1.61M results before I typed anything :(.

I guess the way I'm thinking in my head it would work:
When the page loads:  Do not search or return any results (Assuming SharePoint is default Vertical)
If I type in a word in the search box and press enter or the arrow, search
If I press a refiner.... hmmmm.... would we have the same issue?  Not sure.

I'm just trying to speed up the initial page load because it just takes to long.

I could even go for a toggle in the settings like this:
Do NOT load any SharePoint Results until there is an action from one of the connected webparts.
An action could be Typing a word in Search box, Pressing a vertical.
I am thinking that the Filters webpart would not be visible until there are results because it seems to build it's choices from the results.

Thoughts?

**Reply by wobba** [reply]
You are clearly doing something wrong as setting up the page to show nothing before a user search is pretty straight forward. Try this tutorial https://microsoft-search.github.io/pnp-modern-search/scenarios/page-with-filters/
